client.go

```diff
@@ -256,6 +256,16 @@ func (c *Client) Start() {
 			}
 
 			time.Sleep(dur)
+
+			// exit if closed
+			c.mu.Lock()
```

Comment:
we could change the type of `mu` to `RWMutex` instead of `Mutex`and use RLock and RUnlock here